src/sshfs_mount/sshfs_mount_handler.cpp

```diff
@@ -60,229 +48,162 @@ void start_and_block_until(mp::Process* process, Signal signal, std::function<bo
-            }
-
-            mount_processes[instance].erase(target_path);
-        });
-
-    QObject::connect(
-        sshfs_server_process.get(), &mp::Process::error_occurred, this,
-        [instance = vm->vm_name, target_path, process = sshfs_server_process.get()](QProcess::ProcessError error,
-                                                                                    QString error_string) {
-            mpl::log(mpl::Level::error, category,
-                     fmt::format("There was an error with sshfs_server for instance \"{}\" with path '{}': {} - {}",
-                                 instance, target_path, mp::utils::qenum_to_string(error), error_string));
-        });
+    QObject::connect(process.get(), &Process::finished, [this](const ProcessState& exit_state) {
+        if (exit_state.completed_successfully())
+        {
+            mpl::log(mpl::Level::info, category,
+                     fmt::format("Mount \"{}\" in instance '{}' has stopped", target, vm->vm_name));
+        }
+        else
+        {
+            // not error as it failing can indicate we need to install sshfs in the VM
+            mpl::log(mpl::Level::warning, category,
+                     fmt::format("Mount \"{}\" in instance '{}' has stopped unexpectedly: {}", target, vm->vm_name,
+                                 exit_state.failure_message()));
+        }
+    });
+    QObject::connect(process.get(), &Process::error_occurred, [this](auto error, auto error_string) {
+        mpl::log(mpl::Level::error, category,
+                 fmt::format("There was an error with sshfs_server for instance '{}' with path \"{}\": {} - {}",
+                             vm->vm_name, target, utils::qenum_to_string(error), error_string));
+    });
 
-    mpl::log(mpl::Level::info, category,
-             fmt::format("process program '{}'", sshfs_server_process->program().toStdString()));
-    mpl::log(mpl::Level::info, category,
-             fmt::format("process arguments '{}'", sshfs_server_process->arguments().join(", ").toStdString()));
+    mpl::log(mpl::Level::info, category, fmt::format("process program '{}'", process->program()));
+    mpl::log(mpl::Level::info, category, fmt::format("process arguments '{}'", process->arguments().join(", ")));
 
-    start_and_block_until(
-        sshfs_server_process.get(), &mp::Process::ready_read_standard_output, [](mp::Process* process) {
-            return process->read_all_standard_output().contains("Connected"); // Magic string printed by sshfs_server
-        });
+    start_and_block_until_connected(process.get());
 
     // Check in case sshfs_server stopped, usually due to an error
-    auto process_state = sshfs_server_process->process_state();
+    auto process_state = process->process_state();
     if (process_state.exit_code == 9) // Magic number returned by sshfs_server
-    {
-        throw mp::SSHFSMissingError();
-    }
+        throw SSHFSMissingError();
```

Comment:
OK, I am confused... Didn't we just install it?

Reply:
OK, if I am seeing it right, we should not get into this situation anymore and we could replace this with an `assert(process_state.exit_code != 9)`. However, that should have been the case before this PR and I guess it's out of scope, so feel free to ignore.

Reply:
Again, yeah just ignore.